src/test/java/co/rsk/bitcoinj/script/RedeemScriptParserFactoryTest.java

```diff
@@ -70,14 +76,29 @@ public void get_whenP2shOutputScript_shouldThrowScriptException() {
 
     @Test
     public void get_whenFlyoverStandardRedeemScript_shouldReturnRedeemScriptParser() {
-        byte[] data = Sha256Hash.of(new byte[]{1}).getBytes();
-        Script fastBridgeRedeemScript = RedeemScriptUtils.createFastBridgeRedeemScript(
-            data,
-            defaultRedeemScriptKeys
+        Script redeemScript = RedeemScriptUtils.createStandardRedeemScript(defaultRedeemScriptKeys);
+        Script flyoverRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(
+            FLYOVER_DERIVATION_HASH,
+            redeemScript
         );
 
-        RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(fastBridgeRedeemScript.getChunks());
-        Assert.assertEquals(MultiSigType.FAST_BRIDGE_MULTISIG, redeemScriptParser.getMultiSigType());
+        RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(flyoverRedeemScript.getChunks());
+
+        Assert.assertTrue(redeemScriptParser instanceof FlyoverRedeemScriptParser);
+        assertInternalRedeemScriptParser((FlyoverRedeemScriptParser) redeemScriptParser, StandardRedeemScriptParser.class, STANDARD_MULTISIG);
+        Assert.assertEquals(MultiSigType.FLYOVER, redeemScriptParser.getMultiSigType());
+    }
+
+    private void assertInternalRedeemScriptParser(FlyoverRedeemScriptParser flyoverRedeemScriptParser, Class<?> expectedInternalRedeemScripParser, MultiSigType expectedType) {
+        try {
+            Field internalRedeemScriptParserField = flyoverRedeemScriptParser.getClass().getDeclaredField("internalRedeemScriptParser");
```

Comment:
Is this a good idea? Can't we test this through the public methods?

Reply:
This is just an extra assertion to really ensure the FlyoverRedeemScript is building the right internal redeem script. I don't see a different way on how to do this.